libp2p/pubsub/gossipsub.py

```diff
@@ -215,44 +215,47 @@ def _get_peers_to_send(
         :param origin: peer id of the peer the message originate from.
         :return: a generator of the peer ids who we send data to.
         """
-        send_to: List[ID] = []
+        send_to: Set[ID] = set()
         for topic in topic_ids:
             if topic not in self.pubsub.peer_topics:
                 continue
 
             # floodsub peers
             # FIXME: `gossipsub.peers_floodsub` can be changed to `gossipsub.peers` in go.
             #   This will improve the efficiency when searching for a peer's protocol id.
-            floodsub_peers: List[ID] = [
+            floodsub_peers: Set[ID] = set(
                 peer_id
                 for peer_id in self.pubsub.peer_topics[topic]
                 if peer_id in self.peers_floodsub
-            ]
+            )
+            send_to.update(floodsub_peers)
 
             # gossipsub peers
-            gossipsub_peers: List[ID] = []
+            gossipsub_peers: Set[ID] = set()
             if topic in self.mesh:
                 gossipsub_peers = self.mesh[topic]
             else:
                 # When we publish to a topic that we have not subscribe to, we randomly pick
                 # `self.degree` number of peers who have subscribed to the topic and add them
                 # as our `fanout` peers.
                 topic_in_fanout: bool = topic in self.fanout
-                fanout_peers: List[ID] = self.fanout[topic] if topic_in_fanout else []
+                fanout_peers: Set[ID] = self.fanout[topic] if topic_in_fanout else set()
                 fanout_size = len(fanout_peers)
                 if not topic_in_fanout or (
                     topic_in_fanout and fanout_size < self.degree
                 ):
                     if topic in self.pubsub.peer_topics:
                         # Combine fanout peers with selected peers
-                        fanout_peers += self._get_in_topic_gossipsub_peers_from_minus(
-                            topic, self.degree - fanout_size, fanout_peers
+                        fanout_peers.update(
+                            self._get_in_topic_gossipsub_peers_from_minus(
+                                topic, self.degree - fanout_size, list(fanout_peers)
```

Comment:
Instead of performing `list` on `self.mesh[topic]`, does it make sense to change  `minus: Sequence`  in `_get_in_topic_gossipsub_peers_from_minus` to `Iterable` and pass `self.mesh[topic]` directly?